### ISSUE TITLE
Remember channels that I haven't left by myself & allow listing/enabling them ZNC style

### ISSUE THREAD

**Issue by Mikaela** [matrix-org/matrix-appservice-irc#696]
[ZNC](https://wiki.znc.in/) is the most popular IRC bouncer. It remembers the channels that I join (especially when the [chansaver](https://wiki.znc.in/chansaver) module is loaded) until I `PART` them by hand even if it cannot join the channel (due to join throtthling, not being identified or anything).

When ZNC cannot join a channel automatically, it will try a few times and if it fails, it will just mark the channel as disabled and `*status` (roughly equivalent to matrix-appservice-irc admin room) will warn about it. When I come back and identify to NickServ by myself, I simply issue `/znc enablechan *` or `/msg *status znc enablechan *` and ZNC will say "enabled X channels" and attempt to join the rooms again.

I got this idea as OFTC bridge just reconnected me and sent me 5 errors about 2 rooms (a separate issue?) and the ZNC behaviour would be more easy than rejoining everywhere by hand again, especially if I had more rooms. I don't know what would be a good command for matrix-appservice-irc, but I think the result could be it attempting to join my ghost everywhere and if it succeeds, invite me to the rooms like I had ran `!join #channel`.

I think in addition to OFTC (how come there is no issue on certfp?) this could help people using bridges to other networks that don't support PASS or SASL or that use Anope IRC Services which like OFTC doesn't allow users to identify as other nickname than they are currently using.

**Comment by Mikaela** [comment]
> I got this idea as OFTC bridge just reconnected me and sent me 5 errors about 2 rooms (a separate issue?)

As Riot/flatpak doesn't want to upload this screenshot, I will put it here for now.

![screenshot from 2018-09-27 18-31-23](https://user-images.githubusercontent.com/831184/46157210-e3041500-c283-11e8-85ad-9247078a7020.png)

**Comment by Half-Shot** [comment]
@Mikaela I'm assuming it kicked you (it should have).

For the whole retaining channels that you've left via a bridge issued kick, we could definitely do that.

**Comment by Mikaela** [comment]
Yes, I think it kicked me. And I made that issue text very unclear, TL;DR would be "remember where I am kicked from and re-invite me when I tell the bridge `!rejoin *` or similar".

**Comment by Half-Shot** [comment]
Sounds extremely reasonable to me :)
